### ISSUE TITLE
Broken hyperlinks in resulting makefiles

### ISSUE THREAD

**Issue by kuredoro** [standardese/standardese#189]
At work I wrote a little script that runs a markdown server that serves standardese-generated documentation. The way all the markdown servers work is that they generate an `id` attribute based on the contents of a header. For instance:
```
# Hello
## hello world
```
will generate
```html
<h1 id="hello">Hello</h1>
<h2 id="hello-world">hello world</h2>
```

But standardese generates wrong anchors that refer to entities in another files.

A simple header file I took from another issue here. I named it `test.h`
```cpp
#include <type_traits>

namespace test
{

    /// \effects Exchanges values stored in two locations.
    /// \requires Type `T` shall be `MoveConstructible` and `MoveAssignable`.
    template <class T>
    void swap(T &a, T &b) noexcept(std::is_nothrow_move_constructible<T>::value &&
                                    std::is_nothrow_move_assignable<T>::value);
}
```
Standardese generates the following files: `doc_test.md`, `standardese_entities.md `, `standardese_files.md`,  `standardese_modules.md`. The last file is empty. I'm interested in `standardese_entities.md` but both `standardese_entities.md` and `standardese_files.md` generate wrong hyperlinks.

doc_test.md (``` is replaced by ``)
```
# Header file `test.h`

`` cpp
namespace test
{
    template <class T>
    void swap(T& a, T& b) noexcept(std::is_nothrow_move_constructible<T>::value && std::is_nothrow_move_assignable<T>::value);
}
``

### Function `test::swap`

`` cpp
template <class T>
void swap(T& a, T& b) noexcept(std::is_nothrow_move_constructible<T>::value && std::is_nothrow_move_assignable<T>::value);
``

*Effects:* Exchanges values stored in two locations.

*Requires:* Type `T` shall be `MoveConstructible` and `MoveAssignable`.

-----
```
standardese_entities.md
```
# Project index

  - ## Namespace `test`

      - [`swap`](doc_test.md#standardese-test__swap-T--T--T--)
```

The hyperlink is `doc_test.md#standardese-test__swap-T--T--T--` but in the `doc_test.md` the correct anchor is `#function-test-swap`

Likewise, `standardese_files.md` exhibit the same behavior. Its contents are
```
# Project files

  - [`test.h`](doc_test.md#standardese-test-h)
```

The anchor is `#standardese-test-h` but should be `#header-file-test-h`

**Comment by saraedum** [comment]
In the default setup, standardese generates links but does not generate the anchors. That's unfortunate or at least not well communicated by the tool.

Have you tried running standardese with `--output.format commonmark_html` so it creates the anchors?

This will be fixed in the upcoming version of standardese btw. Unless told otherwise, we then generate links that are compatible with the way mkdocs generates anchors from headings; which is compatible with what you are describing. (I have not looked into how jekyll and others do it exactly yet…)

**Comment by kuredoro** [comment]
Cool that it will be fixed!

So, I've run with `--output.format commonmark_html` and it produced the anchors and I could jump to them. So that kinda fixes my problem. One quirk that I noticed, though, is that Firefox doesn't center the page on the anchored content, so that it appears on top of the page. But since anchors in the produced markdown were just after the headings, the headings themselves were out of the page! It took me a while to realize why it *centers* on another function... 

That is, the produced markdown has something like this
```
...
### Function `test::swap`

<span id="standardese-test__swap-T--T--T--"></span>
...
```
And the "Function `test::swap`" heading is being left out of the page.

**Comment by saraedum** [comment]
I am glad it (mostly) worked. Yes, the `span` is not really in the correct place.

**Comment by caesay** [comment]
Just as one additional datapoint, I am using Docusaurus, and `--output.format commonmark_html` doesn't really work because the anchors are now emitted as html. Docusarus checks for broken links/anchors during the build. It doesn't parse the html, so fails the build thinking that the links are broken.
